pkg/kubernetes/kubernetes.go

```diff
@@ -336,15 +368,43 @@ func CreateCustomResourceDefinition(ctx context.Context, crdName string, crdSing
 
 	// If there is an error, delete the object to keep it clean.
 	if err != nil {
-		log.Infof("Cleanup %q CRD because the CRD created was not successfully established. Error: %+v", crdName, err)
-		deleteErr := apiextensionsClientSet.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(ctx, crdName, metav1.DeleteOptions{})
+		log.Infof("Cleanup %q CRD because the CRD created was not successfully established. Err: %+v", crdName, err)
+		deleteErr := clientSet.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(ctx, crdName, *metav1.NewDeleteOptions(0))
```

Comment:
what is the diff between `metav1.DeleteOptions{}` and `*metav1.NewDeleteOptions(0)`?

Reply:
`*metav1.NewDeleteOptions(0)` allows us to specify a grace period, in this case 0, which means Delete the object immediately.